### PR TITLE
Give more explanatory error for bad 'Mesh/block' specification

### LINKED ISSUE #22928: Set coordinate system error
## Reason
The set coordinate system stuff (related to multiapp transforms) broke a contact tutorial here: `moose/modules/combined/tutorials/introduction/thermal_mechanical_contact/thermomech_cont_step01.i`  (lesson learned: don't just do `check_input = true` testing!)

The error for `thermomech_cont_step01.i` is

```
*** ERROR ***
The following error occurred in the object "mesh", of type "MeshGeneratorMesh".

Requested subdomain 1 does not exist.
```

`Requested subdomain 1 does not exist.`  is confusing. This is a lower-d subdomain created by the contact action. It it referenced nowhere.

## Design

Make this work again (with a `block = 0` in `GlobalParams` to avoid doing mechanics on the lower-d subdomain)

## Impact
Working tutorial

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #22928

### CODE REVIEW

**Comment by lindsayad** [comment]
Ok @dschwen I was able to make your tutorial inputs work without requiring any modification

**Comment by lindsayad** [comment]
@dschwen would you be willing to review this?

File: framework/src/problems/FEProblemBase.C [hunk 1]
```diff
@@ -177,7 +177,7 @@ FEProblemBase::validParams()
   /// 2. _blocks.size() > 0 and no coordinate system was specified, then the whole domain will be XYZ.
   /// 3. _blocks.size() > 0 and one coordinate system was specified, then the whole domain will be that system.
   params.addDeprecatedParam<std::vector<SubdomainName>>(
-      "block", "Block IDs for the coordinate systems", "Please use 'Mesh/block' instead");
+      "block", "Block IDs for the coordinate systems", "Please use 'Mesh/coord_block' instead");
```
Comment (dschwen) [inline framework/src/problems/FEProblemBase.C:180]: Are GlobalParams still applied to deprecated params? If so then this PR will not directly fix my issues, but once the deprecated parameter is removed it should

**Review by dschwen — APPROVED** [review]

File: framework/src/mesh/MooseMesh.C [hunk 2]
```diff
@@ -3523,72 +3535,62 @@ MooseMesh::setCoordSystem(const std::vector<SubdomainName> & blocks,
...
 
-    for (const auto & sbd : subdomains)
-      _coord_sys[sbd] = coord_type;
+  if (coord_sys.size() <= 1)
+  {
+    // We will specify the same coordinate system for all blocks
+    const auto coord_type = coord_sys.size() == 0
+                                ? Moose::COORD_XYZ
+                                : Moose::stringToEnum<Moose::CoordinateSystemType>(coord_sys[0]);
+    for (const auto sid : subdomains)
```
Comment (lindsayad) [inline framework/src/mesh/MooseMesh.C:3574]: @dschwen this fixes your issue regardless of the fact that `block` gets picked up from `GlobalParams`. As long as `coord_sys` is size 1 (or size 0) then we have changed to automatically set *all* mesh subdomains to that coordinate system type, regardless of the value of `block`. Before we just set it for the specifically set blocks

**Pull request merged**